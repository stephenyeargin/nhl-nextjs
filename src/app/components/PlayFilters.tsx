'use client';

import React from 'react';
import PeriodSelector from './PeriodSelector';

interface PlayFiltersProps {
  periodData: any;
  activePeriod: number;
  onPeriodChange: (_n: number) => void;
  includeAll?: boolean;
  eventFilter: string | null;
  onEventFilterChange: (_v: string) => void;
  teamFilter: string | null;
  onTeamFilterChange: (_v: string) => void;
  awayTeam: any;
  homeTeam: any;
}

const PlayFilters: React.FC<PlayFiltersProps> = ({
  periodData,
  activePeriod,
  onPeriodChange,
  includeAll = true,
  eventFilter,
  onEventFilterChange,
  teamFilter,
  onTeamFilterChange,
  awayTeam,
  homeTeam,
}) => {
  return (
    <div className="flex justify-center items-center my-5 text-xs md:text-sm">
      <PeriodSelector
        periodData={periodData}
        activePeriod={activePeriod}
        handlePeriodChange={onPeriodChange}
        includeAll={includeAll}
      />
      <div className="mx-2">
        <select
          className="p-2 min-w-[100px] md:min-w-[150px] border rounded text-black dark:text-white bg-inherit"
          value={eventFilter || 'all'}
          onChange={(e) => onEventFilterChange(e.target.value)}
        >
          <option value="all">All Events</option>
          <option value="goal">Goals</option>
          <option value="shot-on-goal">Shots on Goal</option>
          <option value="missed-shot">Missed Shots</option>
          <option value="blocked-shot">Blocked Shots</option>
          <option value="hit">Hits</option>
          <option value="giveaway">Giveaways</option>
          <option value="takeaway">Takeaways</option>
          <option value="delayed-penalty">Delayed Penalties</option>
          <option value="penalty">Penalties</option>
          <option value="faceoff">Faceoffs</option>
          <option value="stoppage">Stoppage</option>
        </select>
      </div>
      <div>
        <select
          className="p-2 min-w-[100px] md:min-w-[150px] border rounded text-black dark:text-white bg-inherit"
          value={teamFilter || 'all'}
          onChange={(e) => onTeamFilterChange(e.target.value)}
        >
          <option value="all">Both Teams</option>
          <option value={awayTeam.id}>{awayTeam.placeName.default}</option>
          <option value={homeTeam.id}>{homeTeam.placeName.default}</option>
        </select>
      </div>
    </div>
  );
};

export default PlayFilters;
