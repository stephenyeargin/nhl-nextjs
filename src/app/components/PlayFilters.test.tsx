import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import PlayFilters from './PlayFilters';

jest.mock('./PeriodSelector', () => ({
  __esModule: true,
  default: ({ activePeriod, handlePeriodChange }: any) => (
    <button data-testid="period-selector" onClick={() => handlePeriodChange(2)}>
      Period: {activePeriod}
    </button>
  ),
}));

const awayTeam = { id: 2, placeName: { default: 'Columbus' } };
const homeTeam = { id: 1, placeName: { default: 'Nashville' } };

describe('PlayFilters', () => {
  it('invokes handlers on change and displays selected values', () => {
    const onPeriod = jest.fn();
    const onEvent = jest.fn();
    const onTeam = jest.fn();
    const { asFragment } = render(
      <PlayFilters
        periodData={{}}
        activePeriod={0}
        onPeriodChange={onPeriod}
        includeAll
        eventFilter={'all'}
        onEventFilterChange={onEvent}
        teamFilter={'all'}
        onTeamFilterChange={onTeam}
        awayTeam={awayTeam}
        homeTeam={homeTeam}
      />
    );
    expect(asFragment()).toMatchSnapshot();

    // Period selector
    fireEvent.click(screen.getByTestId('period-selector'));
    expect(onPeriod).toHaveBeenCalledWith(2);

    // Event filter
    fireEvent.change(screen.getByDisplayValue('All Events'), { target: { value: 'goal' } });
    expect(onEvent).toHaveBeenCalledWith('goal');

    // Team filter
    fireEvent.change(screen.getByDisplayValue('Both Teams'), {
      target: { value: String(homeTeam.id) },
    });
    expect(onTeam).toHaveBeenCalledWith(String(homeTeam.id));
  });
});
