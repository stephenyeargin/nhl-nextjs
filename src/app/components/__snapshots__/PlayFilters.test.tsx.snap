// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayFilters invokes handlers on change and displays selected values 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center items-center my-5 text-xs md:text-sm"
  >
    <button
      data-testid="period-selector"
    >
      Period: 0
    </button>
    <div
      class="mx-2"
    >
      <select
        class="p-2 min-w-[100px] md:min-w-[150px] border rounded text-black dark:text-white bg-inherit"
      >
        <option
          value="all"
        >
          All Events
        </option>
        <option
          value="goal"
        >
          Goals
        </option>
        <option
          value="shot-on-goal"
        >
          Shots on Goal
        </option>
        <option
          value="missed-shot"
        >
          Missed Shots
        </option>
        <option
          value="blocked-shot"
        >
          Blocked Shots
        </option>
        <option
          value="hit"
        >
          Hits
        </option>
        <option
          value="giveaway"
        >
          Giveaways
        </option>
        <option
          value="takeaway"
        >
          Takeaways
        </option>
        <option
          value="delayed-penalty"
        >
          Delayed Penalties
        </option>
        <option
          value="penalty"
        >
          Penalties
        </option>
        <option
          value="faceoff"
        >
          Faceoffs
        </option>
        <option
          value="stoppage"
        >
          Stoppage
        </option>
      </select>
    </div>
    <div>
      <select
        class="p-2 min-w-[100px] md:min-w-[150px] border rounded text-black dark:text-white bg-inherit"
      >
        <option
          value="all"
        >
          Both Teams
        </option>
        <option
          value="2"
        >
          Columbus
        </option>
        <option
          value="1"
        >
          Nashville
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;
